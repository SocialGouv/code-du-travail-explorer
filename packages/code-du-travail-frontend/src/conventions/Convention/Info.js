import React from "react";
import PropTypes from "prop-types";
import Link from "next/link";
import styled from "styled-components";
import { theme } from "@socialgouv/react-ui";
import { format, parseISO } from "date-fns";
import { formatIdcc } from "@cdt/data";

const Detail = ({ label, children }) => (
  <StyledDetail>
    <DetailLabel>{label} :</DetailLabel>
    {children}
  </StyledDetail>
);

const Info = ({ convention: { num, title, date_publi, url } }) => (
  <React.Fragment>
    <Detail label="Nom complet">{title}</Detail>
    <Detail label="IDCC">{formatIdcc(num)}</Detail>
    {date_publi && (
      <Detail label="Date d’entrée en vigueur">
        {format(parseISO(date_publi), "dd/MM/yyyy")}
      </Detail>
    )}
    {url && (
      <p data-no-glossary>
        <a target="_blank" rel="noopener noreferrer nofollow" href={url}>
          Voir la convention sur Légifrance
        </a>
      </p>
    )}
    <p data-no-glossary>
      <Link href="/glossaire/convention-collective" passHref>
        <a target="_blank" rel="noopener noreferrer nofollow">
          Qu’est-ce qu’une convention collective&nbsp;?
        </a>
      </Link>
    </p>
  </React.Fragment>
);

const { spacings } = theme;

const StyledDetail = styled.div`
  margin-bottom: ${spacings.medium};
`;

const DetailLabel = styled.div`
  display: inline-block;
  margin-right: 10px;
  font-weight: bold;
`;

Info.propTypes = {
  convention: PropTypes.shape({
    url: PropTypes.string,
    idcc: PropTypes.string,
    date_publi: PropTypes.string,
  }).isRequired,
};

export { Info };
