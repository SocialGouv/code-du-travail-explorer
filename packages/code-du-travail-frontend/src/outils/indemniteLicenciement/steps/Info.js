import React from "react";
import { Container } from "@cdt/ui/";
import { TypeContrat } from "../components/TypeContrat";
import { YesNoQuestion } from "../components/YesNoQuestion";

function validate(values) {
  const errors = {};

  if (values.contrat && values.contrat !== "cdi") {
    errors.contrat = (
      <>
        Vous ne pouvez pas obtenir d’indemnité de licenciement si vous êtes en
        CDD ou en contrat de travail temporaire. Sous certaines conditions, vous
        avez le droit à une&nbsp;
        <a
          href="/fiche-service-public/qui-peut-toucher-la-prime-de-precarite-a-la-fin-dun-contrat-de-travail"
          target="_blank"
          el="noopener noreferrer"
        >
          prime de précarité
        </a>
      </>
    );
  }
  if (values.fauteGrave) {
    errors.fauteGrave = (
      <>
        L’indemnité légale de licenciement n’est pas dûe en cas de faute grave.
        <br />
        Vous reporter à la lettre de notification de licenciement, lorsqu’il est
        invoqué le motif de faute grave doit apparaître précisément dans le
        courrier.
      </>
    );
  }
  return errors;
}

function StepInfo() {
  return (
    <Container>
      <TypeContrat />
      <YesNoQuestion
        name="fauteGrave"
        label="Êtes-vous licencié(e) pour faute grave ou lourde&nbsp;?"
      />
      <YesNoQuestion
        name="inaptitude"
        label="Êtes-vous licencié(e) pour inaptitude suite à accident du travail ou
        maladie professionnelle reconnue&nbsp;?"
      />
    </Container>
  );
}
StepInfo.validate = validate;
export { StepInfo };
