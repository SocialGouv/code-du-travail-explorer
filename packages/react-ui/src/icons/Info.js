import React from "react";

export const Info = () => (
  <svg viewBox="0 0 20 20" fill="none">
    <title>info</title>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M10 20C4.47715 20 0 15.5228 0 10C0 4.47716 4.47715 0 10 0C15.5228 0 20 4.47716 20 10C20 15.5228 15.5228 20 10 20ZM10 18.8889C14.9092 18.8889 18.8889 14.9092 18.8889 10C18.8889 5.0908 14.9092 1.11111 10 1.11111C5.0908 1.11111 1.11111 5.0908 1.11111 10C1.11111 14.9092 5.0908 18.8889 10 18.8889ZM8.88889 9.76752C8.88889 9.15954 9.38207 8.66672 10 8.66672C10.6137 8.66672 11.1111 9.16774 11.1111 9.76752V14.2326C11.1111 14.8406 10.6179 15.3334 10 15.3334C9.38635 15.3334 8.88889 14.8324 8.88889 14.2326V9.76752ZM10 6.66667C10.6137 6.66667 11.1111 6.1692 11.1111 5.55556C11.1111 4.94191 10.6137 4.44444 10 4.44444C9.38635 4.44444 8.88889 4.94191 8.88889 5.55556C8.88889 6.1692 9.38635 6.66667 10 6.66667Z"
      fill="currentColor"
    />
  </svg>
);
