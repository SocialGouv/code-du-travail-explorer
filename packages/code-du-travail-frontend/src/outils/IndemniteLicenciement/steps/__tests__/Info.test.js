import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { StepInfo } from "../Info";
import { Form } from "react-final-form";

function renderForm(data) {
  return render(
    <Form
      validate={StepInfo.validate}
      initialValues={{ ...data }}
      onSubmit={jest.fn()}
    >
      {({ form, handleSubmit }) => (
        <form onSubmit={handleSubmit}>
          <StepInfo form={form} />
          <button data-testid="nextBt">suivant</button>
        </form>
      )}
    </Form>
  );
}

describe("<StepInfo />", () => {
  it("should render", () => {
    const { container } = renderForm({});
    expect(container).toMatchSnapshot();
  });

  it("should display error if contrat cdd", () => {
    const { getByLabelText, getByTestId, getByText } = renderForm({
      contrat: "cdi"
    });
    //trigger validation
    const cdd = getByLabelText(/cdd/i);
    fireEvent.click(cdd);
    const nextBt = getByTestId("nextBt");
    nextBt.click();
    expect(getByText(/vous ne pouvez pas/i)).toMatchSnapshot();
  });

  it("should display error if fauteGrave", () => {
    const { getAllByLabelText, getByTestId, getByText } = renderForm({});
    const [fauteGrave] = getAllByLabelText(/oui/i);
    fireEvent.click(fauteGrave);
    const nextBt = getByTestId("nextBt");
    nextBt.click();
    expect(getByText(/n’est pas dûe/i)).toMatchSnapshot();
  });
});
