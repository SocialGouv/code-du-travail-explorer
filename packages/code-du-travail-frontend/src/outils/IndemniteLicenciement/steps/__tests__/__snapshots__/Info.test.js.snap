// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInfo /> should display error if contrat cdd 1`] = `
.c0 {
  margin: 1rem 0 1.6rem;
  color: #eb5757;
  font-weight: 600;
  font-size: 1.4rem;
}

<div
  class="c0"
>
  L’indemnité de licenciement n’est pas dûe pour les CDD et contrats de travail temporaires. Sous certaines conditions, le salarié peut avoir le droit à une 
  <a
    el="noopener noreferrer"
    href="/fiche-service-public/qui-peut-toucher-la-prime-de-precarite-a-la-fin-dun-contrat-de-travail"
    target="_blank"
  >
    indemnité de précarité
  </a>
  .
</div>
`;

exports[`<StepInfo /> should display error if fauteGrave 1`] = `
.c0 {
  margin: 1rem 0 1.6rem;
  color: #eb5757;
  font-weight: 600;
  font-size: 1.4rem;
}

<div
  class="c0"
>
  L’indemnité légale de licenciement n’est pas dûe en cas de faute grave.
  <br />
  Lorsqu’il est invoqué, le motif de faute grave doit apparaître précisément dans le courrier. Reportez vous à la lettre de notification de licenciement.
</div>
`;

exports[`<StepInfo /> should render 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c0 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c1 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

<div>
  <form>
    <p
      class="c0"
    >
      Quel est le type du contrat de travail ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          name="contrat"
          type="radio"
          value="cdd"
        />
        <span>
          Contrat à durée determiné (CDD) ou contrat d’intérim
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="contrat"
          type="radio"
          value="cdi"
        />
        <span>
          Contrat à durée indeterminé (CDI)
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      Le licenciement est-il dû à une faute grave (ou lourde) ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          name="fauteGrave"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="fauteGrave"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <p
      class="c0"
    >
      Le licenciement est-il dû à une inaptitude suite à un accident du travail ou maladie professionnelle reconnue ?
      <span
        aria-label="champs obligatoire"
        class="c1"
      >
        *
      </span>
    </p>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          name="inaptitude"
          type="radio"
          value="true"
        />
        <span>
          Oui
        </span>
      </label>
      <label
        class="c3"
      >
        <input
          name="inaptitude"
          type="radio"
          value="false"
        />
        <span>
          Non
        </span>
      </label>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;
